### PR TITLE
chore(core): avoid GC in rolling log writer

### PR DESCRIPTION
**Pull request by puzpuzpuz**
A follow-up for #2740

### CODE REVIEW

**Comment by bluestreak01** [comment]
Excellent, thank you 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 75 / 76 (98.68%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/LogRollingFileWriter.java|62|63|98.41%|
|:large_blue_circle:|io/questdb/std/str/DirectCharSink.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/std/Numbers.java|4|4|100.00%|

File: core/src/main/java/io/questdb/log/LogRollingFileWriter.java [hunk 1]
```diff
@@ -369,69 +376,80 @@ private void pushFileStackUp() {
...
+        long totalSize = 0;
+        for (long i = logFileList.size() - 1; i > -1; i -= 2) {
+            final long packedOffsets = logFileList.get(i);
+            final int startOffset = Numbers.decodeLowInt(packedOffsets);
+            final int endOffset = Numbers.decodeHighInt(packedOffsets);
+            CharSequence fileName = logFileNameSink.subSequence(startOffset, endOffset);
+            path.trimTo(logDir.length()).concat(fileName).$();
+            if ((totalSize += Files.length(path)) > nSizeLimit) {
                 if (!ff.remove(path)) {
                     throw new LogError("cannot remove: " + path.$());
```
Comment (marregui) [inline core/src/main/java/io/questdb/log/LogRollingFileWriter.java:392]: nit: the `.$()` after path, on throw cannot remove, is not necessary.

**Review by marregui — DISMISSED** [review]
LGTM

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 75 / 76 (98.68%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/LogRollingFileWriter.java|62|63|98.41%|
|:large_blue_circle:|io/questdb/std/str/DirectCharSink.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/std/Numbers.java|4|4|100.00%|

File: core/src/main/java/io/questdb/log/LogRollingFileWriter.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/log/LogRollingFileWriter.java]: Addressed in 2d4a4dbecca0bc85bf2d84467faf27ff9ab19287

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 75 / 77 (97.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/LogRollingFileWriter.java|62|64|96.88%|
|:large_blue_circle:|io/questdb/std/str/DirectCharSink.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/std/Numbers.java|4|4|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 75 / 77 (97.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/LogRollingFileWriter.java|62|64|96.88%|
|:large_blue_circle:|io/questdb/std/str/DirectCharSink.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/std/Numbers.java|4|4|100.00%|

**Pull request merged**